### ISSUE TITLE
Why not use add_resource() to add instance of resource

### ISSUE THREAD

**Issue by kevin-leptons** [flask-restful/flask-restful#630]
flask-restful use add_resource() to add resource class to route below

```python
class User(Resource):
    def get(self):
        pass

app = Flask(__name__)
api = Api(app)

api.add_resource(User, ...)
```

So I can not inject database connection into resource class. Why flask-restful do not use add_resource() to add instance of resource. By that way, I can free to create instance of resources

```python
class User(Resource):
    def __init__(self, db):
        self.__db = db

    def get(self):
        return self.__db.find_one()

app = Flask(__name__)
api = Api(app)
mongo_client= MongoClient('...')
db = mongo_client.gwisp

api.add_resource(User(db), ...)
````

**Comment by justanr** [comment]
You can provide resource_class_args/kwargs as keyword arguments when registering a resource, they did be a sequence and mapping respectively. 

As for why not provide an instance, it's so any changes you make to the state of the resource during a request is thrown away after the request. Vanilla Foam does the same thing with its class views

**Comment by kevin-leptons** [comment]
Oh, I got it and I have other question

For each request, flask-restful create an instance of resource and free it after that. That's mean

- All of instance of resource are the same. It is created by same class, same arguments
- Flask have not process request by multi-threading. Memory of instance of resource is safe

So why not use only instance of resource to reduce creation time ?

**Comment by joshfriend** [comment]
> So why not use only instance of resource to reduce creation time ?

There's ongoing discussion about that on #616

**Comment by kevin-leptons** [comment]
Problem is solved by [#616](https://github.com/flask-restful/flask-restful/pull/616)
